HHS/Head-Start-TTADP
# [Prod] Strip whitespace from beginning and end of string when creating RTR objectives, improve RSS feed processing, properly save source on RTR

## Description of change

- RTR goal form did not strip leading or trailing whitespace from objective titles. This can lead to unintentional duplicates
- RTR goal form was incorrectly setting goal source in the frontend. 
- RSS feed processing uses a different approach that honors the markup and layout in confluence

## How to test

- Create a goal/objective on the RTR with extra space at the end of the objective. Use that goal/objective on the AR and confirm on the RTR that there is only one RTR
- Set a goal with root causes/source on the RTR and save it. Confirm that you don't get an error.
- View Confluence help drawers and confirm that the content is readable (it may need reworking to match initial design)

## Issue(s)

* https://jira.acf.gov/browse/TTAHUB-3409
* https://jira.acf.gov/browse/TTAHUB-3364

## Checklists

### Every PR



### Before merge to main


### Production Deploy


### After merge/deploy
